### PR TITLE
mention requirement for outbound traffic for kic+konnect

### PR DESCRIPTION
**Pull request by czeslavo**
### Description

If our users have traffic restrictions of any kind for the outbound traffic, it's worth to mention what hosts they should have allow-listed for KIC+Konnect integration.
 
Closes https://github.com/Kong/kubernetes-ingress-controller/issues/4298.

### Testing instructions

https://deploy-preview-5781--kongdocs.netlify.app/kubernetes-ingress-controller/latest/concepts/deployment/
https://deploy-preview-5781--kongdocs.netlify.app/konnect/runtime-manager/kic/

### Checklist 

- [x] Review label added 
- [x] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)

### CODE REVIEW

File: app/_src/kubernetes-ingress-controller/concepts/deployment.md
Comment (cloudjumpercat) [inline app/_src/kubernetes-ingress-controller/concepts/deployment.md:379]:
```suggestion
As {{site.kic_product_name}} calls {{site.konnect_short_name}}'s APIs, outbound traffic from {{site.kic_product_name}}'s pods must be allowed to reach {{site.konnect_short_name}}'s `*.konghq.com` hosts.
```

File: app/konnect/runtime-manager/kic.md
Comment (cloudjumpercat) [inline app/konnect/runtime-manager/kic.md:47]:
```suggestion
* Because {{site.kic_product_name}} calls {{site.konnect_short_name}}'s APIs, outbound traffic from {{site.kic_product_name}}'s pods must be allowed to reach {{site.konnect_short_name}}'s `*.konghq.com` hosts.
```

Comment (cloudjumpercat) [reply app/konnect/runtime-manager/kic.md:47]: Should we also add this information to https://deploy-preview-5781--kongdocs.netlify.app/konnect/network/ under "Hostnames"?

**Review by cloudjumpercat — COMMENTED** [review]
@czeslavo Thanks for making these updates! I left one question about if we should also add this information to the Ports page in Konnect.

**Review by cloudjumpercat — APPROVED** [review]

**Comment by cloudjumpercat** [comment]
@czeslavo Hold off on merging this, Smriti will merge this once KIC in Konnect is GA.

File: app/konnect/runtime-manager/kic.md
Comment (czeslavo) [reply app/konnect/runtime-manager/kic.md]: Yes, I think that's a good point. Updated the `/konnect/network` accordingly and linked to it from the 2 articles edited.

**Review by cloudjumpercat — APPROVED** [review]
@czeslavo LGTM! Just made some small formatting updates. I see there's an test failure for broken links, but when I looked into it, none of the links are on pages you edited, so I'm not sure why they suddenly failed. I'm going to ignore the link checks and I'll merge this in once KIC in Konnect goes GA. Thanks for making these updates!

**Pull request merged**
